aspdotnet/tests/test_aspdotnet.py

```diff
@@ -0,0 +1,118 @@
+        "_LM_W3SVC_1_ROOT_PowerShell",
+        "_LM_W3SVC_1_ROOT",
+        "_LM_W3SVC_2_ROOT_PowerShell",
+        "_LM_W3SVC_1_ROOT_OAB",
+        "_LM_W3SVC_2_ROOT_owa",
+        "_LM_W3SVC_1_ROOT_owa",
+    )
+
+    def _test_basic_check(self, Aggregator):
+        instance = MINIMAL_INSTANCE
+        c = AspdotnetCheck(self.CHECK_NAME, {}, {}, [instance])
+        c.check(instance)
+
+        for metric in self.ASP_METRICS:
+            Aggregator.assert_metric(metric, tags=None, count=1)
+
+        for metric in self.ASP_APP_METRICS:
+            for i in self.ASP_APP_INSTANCES:
+                Aggregator.assert_metric(metric, tags=["instance:%s" % i], count=1)
+
+        assert Aggregator.metrics_asserted_pct == 100.0
+
+    def test_basic_check(self, Aggregator):
+        mockpdh.initialize_pdh_tests()
+        with mock.patch('_winreg.QueryValueEx', mockpdh.mock_QueryValueEx):
+            with mock.patch('win32pdh.LookupPerfNameByIndex', mockpdh.mock_LookupPerfNameByIndex):
+                with mock.patch('win32pdh.EnumObjectItems', mockpdh.mock_EnumObjectItems):
+                    with mock.patch('win32pdh.MakeCounterPath', mockpdh.mock_MakeCounterPath):
+                        with mock.patch('win32pdh.AddCounter', mockpdh.mock_AddCounter):
+                            with mock.patch('win32pdh.GetFormattedCounterValue', mockpdh.mock_GetFormattedCounterValue):
+                                with mock.patch('win32pdh.CollectQueryData', mockpdh.mock_CollectQueryData):
+                                    self._test_basic_check(Aggregator)
+
+    def _test_with_tags(self, Aggregator):
+        instance = INSTANCE_WITH_TAGS
+        c = AspdotnetCheck(self.CHECK_NAME, {}, {}, [instance])
+        c.check(instance)
+
+        for metric in self.ASP_METRICS:
+            Aggregator.assert_metric(metric, tags=['tag1', 'another:tag'], count=1)
+
+        for metric in self.ASP_APP_METRICS:
+            for i in self.ASP_APP_INSTANCES:
+                Aggregator.assert_metric(metric, tags=['tag1', 'another:tag', "instance:%s" % i], count=1)
+
+        assert aggregator.metrics_asserted_pct == 100.0
+
+    def test_with_tags(self, Aggregator):
+        mockpdh.initialize_pdh_tests()
+        with mock.patch('_winreg.QueryValueEx', mockpdh.mock_QueryValueEx):
+            with mock.patch('win32pdh.LookupPerfNameByIndex', mockpdh.mock_LookupPerfNameByIndex):
+                with mock.patch('win32pdh.EnumObjectItems', mockpdh.mock_EnumObjectItems):
+                    with mock.patch('win32pdh.MakeCounterPath', mockpdh.mock_MakeCounterPath):
+                        with mock.patch('win32pdh.AddCounter', mockpdh.mock_AddCounter):
+                            with mock.patch('win32pdh.GetFormattedCounterValue', mockpdh.mock_GetFormattedCounterValue):
+                                with mock.patch('win32pdh.CollectQueryData', mockpdh.mock_CollectQueryData):
+                                    self._test_with_tags(Aggregator)
```

Comment:
are these mocks the same? You could factor this out into a fixture, like I did here: https://github.com/DataDog/integrations-core/blob/master/istio/tests/test_check.py#L176-L190